packages/fiori/src/ShellBar.js

```diff
@@ -380,16 +380,15 @@ class ShellBar extends UI5Element {
 			},
 		};
 
+		this.menuItemsObserver = new MutationObserver(() => {
+			this._updateClonedMenuItems();
+		});
+
 		this._header = {
 			press: event => {
-				if (this.menuItems.length) {
-					this._menuPopoverItems = [];
-					this.menuItems.forEach(item => {
-						this._menuPopoverItems.push(item.textContent);
-					});
-					this.updateStaticAreaItemContentDensity();
-					this.menuPopover.openBy(this.shadowRoot.querySelector(".ui5-shellbar-menu-button"));
-				}
+				this._updateClonedMenuItems();
+				this.updateStaticAreaItemContentDensity();
+				this.menuPopover.openBy(this.shadowRoot.querySelector(".ui5-shellbar-menu-button"));
```

Comment:
Is it intended that this line is outside of the `if`? So we always open the popup even though it's empty?

Reply:
oh, this is by mistake when I extracted the code to _updateClonedMenuItems